### PR TITLE
Add /api/v1/settings API Endpoint for updating user settings

### LINKED ISSUE #251: Add API endpoint for Settings
**Is your feature request related to a problem? Please describe.**
Related to https://github.com/jointwt/goryon/issues/39
**Describe the solution you'd like**
Create an api endpoint for settings so that users can change their settings through the app (or any api clients)

### PR DESCRIPTION
**Pull request by prologic**
##### Summary

ssia

##### Component Name

- area/backend

##### Test Plan

@dooven to test :)

##### Additional Information

Closes #251

### CODE REVIEW

File: internal/api.go [hunk 1]
```diff
@@ -756,6 +757,80 @@ func (a *API) UnfollowEndpoint() httprouter.Handle {
...
+				log.WithError(err).Error("error updating user avatar")
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
+		}
+
+		user.Email = email
+		user.Tagline = tagline
+
+		user.Theme = theme
```
Comment (dooven) [inline internal/api.go:816]: Not sure how to deal storing & reading the user theme in the app. Just wondering if we can ignore updating the theme right now in the api

Comment (prologic) [reply internal/api.go:816]: Yeah I agree that this was/is really a setting for the Web App. On the mobile app side with Goryon you can just store this in the app's settings in secure storage right? Weren't you also going to change the setting/behavior to match the web app? Auto, Light, Dark? I'll delete this line though...

Comment (prologic) [reply internal/api.go]: Done!

Comment (dooven) [reply internal/api.go]: Yeah I'll do the auto, light and dark settings and move it to the settings page as well!

**Review by dooven — APPROVED** [review]
Lgtm!

**Pull request merged**